applications/mne_x/plugins/babymeg/FormFiles/babymeghpidgl.cpp

```diff
@@ -0,0 +1,199 @@
+//=============================================================================================================
+/**
+* @file     babymeghpidgl.cpp
+* @author   Limin Sun <liminsun@nmr.mgh.harvard.edu>;
+*           Matti Hamalainen <msh@nmr.mgh.harvard.edu>
+* @version  1.0
+* @date     March, 2015
+*
+* @section  LICENSE
+*
+* Copyright (C) 2015, Limin Sun and Matti Hamalainen. All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without modification, are permitted provided that
+* the following conditions are met:
+*     * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+*       following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and
+*       the following disclaimer in the documentation and/or other materials provided with the distribution.
+*     * Neither the name of MNE-CPP authors nor the names of its contributors may be used
+*       to endorse or promote products derived from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+* PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+* INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief    Contains the implementation of the BabyMEGSQUIDControlDgl class.
+*
+*/
+
+#include <QFile>
+#include <QFileDialog>
+#include <QFileInfo>
+
+
+#include "babymeghpidgl.h"
+#include "ui_babymeghpidgl.h"
+
+
+using namespace BabyMEGPlugin;
+
+babymeghpidgl::babymeghpidgl(BabyMEG* p_pBabyMEG,QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::babymeghpidgl),
+    m_pBabyMEG(p_pBabyMEG)
+{
+    ui->setupUi(this);
+
+    connect(ui->bn_PolhemusFile, SIGNAL(released()), this, SLOT(bnLoadPolhemusFile()));
+    connect(this,&babymeghpidgl::SendHPIFiffInfo,m_pBabyMEG,&BabyMEG::RecvHPIFiffInfo);
+    connect(ui->buttonBox,SIGNAL(clicked(QAbstractButton *)),this,SLOT(OKProc(QAbstractButton *)));
+//    connect(ui->buttonBox,SIGNAL(rejected()),this,SLOT(CancelProc));
+}
+
+babymeghpidgl::~babymeghpidgl()
+{
+    delete ui;
+}
+
+void babymeghpidgl::CancelProc()
+{
+
+}
+
+void babymeghpidgl::OKProc(QAbstractButton *b)
+{
+    qDebug()<<"Clicked group button";
+
+    auto s(b->text());
```

Comment:
>> +
>> +babymeghpidgl::~babymeghpidgl()
>> +{
>> +    delete ui;
>> +}
>> +
>> +void babymeghpidgl::CancelProc()
>> +{
>> +
>> +}
>> +
>> +void babymeghpidgl::OKProc(QAbstractButton *b)
>> +{
>> +    qDebug()<<"Clicked group button";
>> +
>> +    auto s(b->text());
>
> please don't use the auto variable

Ok. Please remove auto.




The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.